app/ldap/server/sync.js

```diff
@@ -175,14 +202,122 @@ export function getDataToSyncUserData(ldapUser, user) {
 		return userData;
 	}
 }
+export function mapLdapGroupsToUserRoles(ldapUser, user) {
+	const syncUserRoles = settings.get('LDAP_Sync_User_Data_Groups');
+	const syncUserRolesAutoRemove = settings.get('LDAP_Sync_User_Data_Groups_AutoRemove');
+	const syncUserRolesFieldMap = settings.get('LDAP_Sync_User_Data_GroupsMap').trim();
+
+	const roles = Roles.find({}, {
+		fields: {
+			_updatedAt: 0,
+		},
+	}).fetch();
+	const userRoles = [];
+	if (syncUserRoles && syncUserRolesFieldMap) {
+		const fieldMap = JSON.parse(syncUserRolesFieldMap);
+		_.map(fieldMap, function(userField, ldapField) {
+			const [roleName] = userField.split(/\.(.+)/);
+			if (!_.find(roles, (el) => el._id === roleName)) {
+				logger.debug(`User Role doesn't exist: ${ roleName }`);
+			} else {
+				logger.debug(`User role exists for mapping ${ roleName } -> ${ ldapField }`);
+
+				if (!isUserInLDAPGroup(ldapUser, user, ldapField)) {
```

Comment:
Could invert the logic here to have an if with one statement only and use a return